packages/react/src/tooltip/Tooltip.js

```diff
@@ -74,103 +83,172 @@ const Tooltip = forwardRef((
     if (!isControlled) {
       closeWithDelay();
     }
 
-    if (onCloseProp) {
+    if (typeof onCloseProp === 'function') {
       onCloseProp();
     }
-  };
+  }, [isOpenProp, onCloseProp, closeWithDelay]);
 
-  const handleClick = () => {
+  const handleBlur = handleClose;
+  const handleClick = useCallback(() => {
     if (closeOnClick) {
-      closeWithDelay();
+      handleClose();
     }
-  };
+  }, [closeOnClick, handleClose]);
+  const handleFocus = handleOpen;
+  const handleKeyDown = useCallback((event) => {
+    if (isOpen && closeOnEsc && event.key === 'Escape') {
+      handleClose();
+    }
+  }, [isOpen, closeOnEsc, handleClose]);
+  const handleMouseDown = useCallback(() => {
+    if (closeOnMouseDown) {
+      handleClose();
+    }
+  }, [closeOnMouseDown, handleClose]);
+  const handleMouseEnter = handleOpen;
+  const handleMouseLeave = handleClose;
 
-  const arrowSize = '6px';
-  const tooltipStyleProps = useTooltipStyle();
-  const getTooltipTriggerProps = () => {
-    const tooltipTriggerStyleProps = {
-      display: 'inline-flex',
+  useEventListener(
+    () => ownerDocument(anchorRef.current),
+    'keydown',
+    closeOnEsc ? handleKeyDown : undefined,
+  );
+
+  /**
+   * This allows for catching the "mouseleave" event when the tooltip trigger is disabled.
+   * There is currently a known issue in React regarding the onMouseLeave polyfill.
+   * @see https://github.com/facebook/react/issues/11972
+   */
+  useEventListener(
+    () => anchorRef.current,
+    'mouseleave',
+    handleMouseLeave,
+  );
+
+  useEffect(() => {
+    return () => {
+      if (enterTimeoutRef.current) {
+        clearTimeout(enterTimeoutRef.current);
+        enterTimeoutRef.current = null;
+      }
+      if (exitTimeoutRef.current) {
+        clearTimeout(exitTimeoutRef.current);
+        exitTimeoutRef.current = null;
+      }
     };
+  }, []);
+
+  const arrowSize = '6px'; // FIXME: this should be a theme value
+  const tooltipStyleProps = useTooltipStyle();
+  const getTooltipTriggerProps = (ownProps = {}, ownRef = null) => {
     const eventHandlerProps = {
-      onMouseEnter: handleOpen,
-      onMouseLeave: handleClose,
-      onClick: handleClick,
-      onFocus: handleOpen,
-      onBlur: handleClose,
+      onBlur: wrapEvent(ownProps?.onBlur, handleBlur),
+      onClick: wrapEvent(ownProps?.onClick, handleClick),
+      onFocus: wrapEvent(ownProps?.onFocus, handleFocus),
+      onMouseDown: wrapEvent(ownProps?.onMouseDown, handleMouseDown),
+      onMouseEnter: wrapEvent(ownProps?.onMouseEnter, handleMouseEnter),
     };
 
     return {
-      'aria-describedby': _isOpen ? tooltipId : undefined,
-      ref: combinedRef,
-      role: 'presentation',
-      ...tooltipTriggerStyleProps,
+      ...ownProps,
+      'aria-describedby': isOpen ? tooltipId : undefined,
+      ref: mergeRefs(combinedRef, ownRef),
       ...eventHandlerProps,
     };
   };
 
+  const trigger = (() => {
```

Comment:
No sure if we could use 'useMemo' here

Reply:
I don't think it's necessary to use `useMemo` for the trigger, but it's worth memozing the return value of the `getTooltipTriggerProps()` function if possible.

Reply:
Get it, thanks for your replying~

Another perspective, maybe put trigger to include `children({ getTooltipTriggerProps })`? Coz currently `trigger` would run no matter children type, but that's not what we want.
```
const trigger = useMemo(() => {
  if (typeof children === 'function') {
    return children({ getTooltipTriggerProps });
  }

  if (shouldWrapChildren) {
    return (
      <Box
        display="inline-flex"
        tabIndex={0}
        {...getTooltipTriggerProps()}
      >
        {children}
      </Box>
    );
  }

  // Ensure tooltip has only one child node
  const child = React.Children.only(children);
  return cloneElement(child, getTooltipTriggerProps(child?.props, child?.ref));
  );
}, [/*plz help to fill all deps XD*/]);
```

Reply:
When using FaCC, you should choose `useCallback` rather than `useMemo` because we never know what will be returned by the children function.

Reply:
`useMemo` was added in https://github.com/trendmicro-frontend/tonic-ui/pull/555/commits/f9e3b488b16779ef86339bd58f74edc0b10f33e8

Reply:
https://javascript.plainenglish.io/can-usememo-skip-a-child-render-94e61f5ad981

Reply:
@light049 Please help take a look at the new change. Thank you!